doc/design/audit-trail-mvp.md

```diff
@@ -0,0 +1,156 @@
+# Audit Trail MVP
+
+NIH has required that DANDI Archive generate audit logs while leaving undefined
+what that means. We are choosing to fulfill the requirement with a feature that
+indicates to Dandiset owners and admins the sequence of significant actions
+taken to modify Dandisets, along with identification of who performed each
+action.
+
+The major motivation is to inform owners about how their Dandisets have changed
+when a given owner did not make the change and wishes to know who to speak to
+about those changes. In essence, it is a tool to support effective
+collaboration.
+
+## Requirements
+
+1. **Persistent tracking of significant actions.** Whenever a *significant
+   action* (see list [below](#significant-actions)) occurs successfully, the archive will generate a
+   persistent record of that action, including the following data:
+    - The date and time
+    - The authorized user who triggered the action (note that this may be an actual user or the system itself for internal automated actions)
+    - The ID of the Dandiset on which the action occurs
+    - Event-specific data about the action itself
+
+2. **Display of audit log (to owners and admins only).** The Dandiset landing
+   page will display a rendered (i.e. human-readable) version of the log event
```

Comment:
```suggestion
   page will provide ability to display a rendered (i.e. human-readable) version of the log event
```

since I do not think we want to actually display the log, may be just the last action associated with the dandiset, clearly not more than that. But we would want some interface (e.g. a modal window) to display/navigate and potentially search

Reply:
Why do you think we don't want to display the log? If you read the motivation section, you'll see that we're conceiving of this as an aid to Dandiset owners (to see e.g. who made specific changes to the Dandiset, etc.), which would require displaying the information. This reflects Satra's comments and the sample log display he mocked up.

Reply:
because it would clutter the display and generally would not be of interest to anyone. For the same reason github repository page does not list you the log -- you need to click on "commits" or alike to get to it.  But GitHub does display the latest day an item was modified at to give user ability to assess how stale the project is.  Hence my motivation to not bother displaying the full log but rather state the last modification date and give ability to list entire log in some modal window.

But anyways -- feel free to ignore since it is a minor and UX aspect.

PS I missed the sample log display I guess.

Reply:
> generally would not be of interest to anyone

I am really surprised to see this--the entire motivation for this was to provide helpful information to Dandiset owners (again, see the motivation section of the doc). If that's not the case, why are we even implementing this?

Because I'm getting super mixed signals about this, I'd like to get @satra's take. If we don't want to include the display in the frontend, I'm happy to remove that from the requirements, but I really thought that was a goal of this feature.

> list entire log in some modal window

This would be the design approach. I was not planning on including the full log in the DLP or something like that.

Reply:
nevermind @waxlamp - we can treat my suggestion as a nit pick and UI design specific. I only argued about how it would be communicated, not that it would not communicated at all.